### PR TITLE
chore: add OpentelemetryLogBuilder::with_labels()

### PR DESCRIPTION
**Pull request by andylokandy**

### CODE REVIEW

File: src/append/opentelemetry.rs [hunk 1]
```diff
@@ -80,16 +80,37 @@ impl OpentelemetryLogBuilder {
     }
 
     /// Add a label to the resource.
-    pub fn add_label(mut self, key: impl Into<Cow<'static, str>>, value: impl Into<Cow<'static, str>>) -> Self {
+    pub fn with_label(
```
Comment (tisonkun) [inline src/append/opentelemetry.rs:83]: IIRC in builder pattern, `add_` means add on current state, while `with_` means replace current state.

So I'd prefer `add_label` and `add_labels`.

Comment (andylokandy) [reply src/append/opentelemetry.rs:83]: but `add_xxx` is rust usually means `fn add_xxx(&mut self, x)`

Comment (tisonkun) [reply src/append/opentelemetry.rs:83]: https://users.rust-lang.org/t/next-steps-in-builder-pattern/90341 adopts an `add_` flavor.

While [`derive_builder`](https://crates.io/crates/derive_builder) use `xxx` directly without any prefix. That looks reasonable to me too. And thus:

```rust
pub fn label(mut self, key, value);
pub fn labels(mut self, labels: impl ...);
```

**Comment by tisonkun** [comment]
@andylokandy would you move this PR forward recently?

File: src/append/opentelemetry.rs (hunk 1 above)
Comment (tisonkun) [reply src/append/opentelemetry.rs:83]: Updated.

**Review by tisonkun — APPROVED** [review]

**Pull request merged**